backend/src/main/kotlin/app/ehrenamtskarte/backend/matomo/Matomo.kt

```diff
@@ -0,0 +1,159 @@
+import app.ehrenamtskarte.backend.config.ProjectConfig
+import app.ehrenamtskarte.backend.stores.webservice.schema.SearchParams
+import app.ehrenamtskarte.backend.verification.database.CodeType
+import app.ehrenamtskarte.backend.verification.database.repos.CardRepository
+import app.ehrenamtskarte.backend.verification.webservice.schema.types.CardGenerationModel
+import jakarta.servlet.http.HttpServletRequest
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.matomo.java.tracking.MatomoRequest
+import org.matomo.java.tracking.MatomoRequestBuilder
+import org.matomo.java.tracking.MatomoTracker
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.util.concurrent.ExecutionException
+
+enum class EventCategory {
+    Cards
+}
+
+object Matomo {
+    val logger = LoggerFactory.getLogger(Matomo::class.java)
+    private const val ACTION_NAME = "Request"
+
+    private fun sendTrackingRequest(matomoConfig: MatomoConfig, requestBuilder: MatomoRequestBuilder) {
+        val siteId = matomoConfig.siteId
+        val url = matomoConfig.url
+        val tracker = MatomoTracker(url)
+
+        val matomoRequest = requestBuilder
+            .actionName(ACTION_NAME)
+            .siteId(siteId)
+            .authToken(matomoConfig.accessToken)
+            .build()
+
+        try {
+            val response = tracker.sendRequestAsync(matomoRequest)
+            val httpResponse = response.get()
+            if (httpResponse.statusLine.statusCode > 399) {
+                logger.debug("Request failed with status code ${httpResponse.statusLine.statusCode}")
+            }
+        } catch (e: Exception) {
+            when (e) {
+                is IOException -> logger.debug("Could not send request to Matomo")
+                is ExecutionException, is InterruptedException ->
+                    logger.debug("Error while getting response")
+            }
+        }
+    }
+
+    private fun sendBulkTrackingRequest(matomoConfig: MatomoConfig, requestBuilder: Iterable<MatomoRequestBuilder>) {
+        val siteId = matomoConfig.siteId
+        val url = matomoConfig.url
+        val tracker = MatomoTracker(url)
+        val matomoRequests = requestBuilder.map {
+            it.actionName(ACTION_NAME)
+            it.siteId(siteId)
+            it.authToken(matomoConfig.accessToken)
+            it.build()
+        }
+        try {
+            val response = tracker.sendBulkRequestAsync(matomoRequests)
+            val httpResponse = response.get()
+            if (httpResponse.statusLine.statusCode > 399) {
+                logger.debug("Request failed with status code ${httpResponse.statusLine.statusCode}")
+            }
+        } catch (e: Exception) {
+            when (e) {
+                is IOException -> logger.debug("Could not send request to Matomo")
+                is ExecutionException, is InterruptedException ->
+                    logger.debug("Error while getting response")
```

Comment:
Any recommendation on better loggings are welcome :)

Reply:
for me its fine